docs(store): add guide for provideMockStore (#1591)
ngrx/platform: projects/ngrx.io/content/guide/store/testing.md

# Testing

### Using a Mock Store

> The `provideMockStore()` method registers providers that allow you to mock out the `Store` for testing functionality that has a dependency on `Store` without setting up reducers. 

Author: I think this is not a `method` but just `function`. isn't?

Reviewer: Yep, `function` is better.

  });
});
</code-example>

> ### provideMockStore

Reviewer: ```suggestion
### Using a Mock Store
```

> Provides a *mock store* for testing effects or something which has a dependency on `Store` without setting up reducers. 

Reviewer: ```suggestion
The `provideMockStore()` method registers providers that allow you to mock out the `Store` for testing functionality that has a dependency on `Store` without setting up reducers. 
```

You can write tests validating behaviors corresponding to the specific state snapshot easily.

> Note: All dispatched actions don't affect to the state, but you can see them in the `Actions` stream.

Reviewer: ```suggestion
<div class="alert is-helpful">

**Note:** All dispatched actions don't affect to the state, but you can see them in the `Actions` stream.

</div>
```

Usage: 

<code-example header="auth.guard.spec.ts">
import { TestBed } from '@angular/core/testing';

> import { provideMockActions } from '@ngrx/effects/testing';

Reviewer: Remove unused imports

import { Store } from '@ngrx/store';
import { provideMockStore } from '@ngrx/store/testing';
import { cold } from 'jasmine-marbles';

import { AuthGuard } from '../guards/auth.guard';
import * as AuthActions from '../actions/auth-actions';

describe('Auth Guard', () => {
  let guard: AuthGuard;

>   let mockStore: MockStore<{ loggedIn: boolean } >;

Reviewer: ```suggestion
  let store: MockStore<{ loggedIn: boolean } >;
```

>   let mockStore: MockStore<{ loggedIn: boolean } >;

Reviewer: The greater than/less than arrows need to be HTML entities to display correctly

  const initialState = { loggedIn: false };

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        // any modules needed
      ],
      providers: [
        AuthGuard,
        provideMockStore({ initialState }),
        // other providers
      ],
    });

    guard = TestBed.get(AuthGuard);

>     mockStore = TestBed.get(Store);

Reviewer: ```suggestion
    store = TestBed.get(Store);
```

  });

  it('should return false if the user state is not logged in', () => {
    const expected = cold('(a|)', { a: false });

    expect(guard.canActivate()).toBeObservable(expected);
  });

  it('should return true if the user state is logged in', () => {

>     mockStore.setState({ loggedIn: true });

Reviewer: ```suggestion
    store.setState({ loggedIn: true });
```
